modules/sdk-lib-mpc/src/tss/ecdsa/noSmallFactorsProof.ts

```diff
@@ -0,0 +1,145 @@
+/**
+ * Implementation of No Small Factors ($\Pi^\text{fac}).
+ * https://eprint.iacr.org/2020/492.pdf Section B.4
+ */
+import { createHash } from 'crypto';
+import { bitLength, randBetween } from 'bigint-crypto-utils';
+import { modPow } from 'bigint-mod-arith';
+import { bigIntFromBufferBE, bigIntToBufferBE } from '../../util';
+import { DeserializedNoSmallFactorsProof } from './types';
+
+const ORDER = BigInt('0xfffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364141');
+const ELL = BigInt(256);
+const EPSILON = BigInt(BigInt(2) * ELL);
```

Comment:
Is the choice of `EPSILON` assuming the bit length of paillier modulus `N` is 2048, since at the end of the proof from the paper: assuming $2^{2l+\epsilon} \approx \sqrt{N}$?

Reply:
ELL is defined as 256 assuming a modulus bit length of 3072. I'm not sure why but an ELL of 384 fails the range check part of verification.

Reply:
Maybe the $\approx$ is pretty rough.  Choosing 384 for ELL and therefore EPSILON 768 might cut too close to 3072.